### ISSUE TITLE
Prepare Bromite for the deprecation of third-party cookies

### ISSUE THREAD

**Issue by uazo** [bromite/bromite#2052]
### Preliminary checklist

- [X] I have read the [README](https://github.com/bromite/bromite/blob/master/README.md)
- [X] I have read the [FAQs](https://github.com/bromite/bromite/blob/master/FAQ.md).
- [X] I have searched [existing issues](https://github.com/bromite/bromite/issues) for my feature request. This is a new issue (NOT a duplicate) and is not related to another issue.

### Is your feature request related to privacy?

Yes

### Is there a patch available for this feature somewhere?

No

### Describe the solution you would like

since it seems that CHIPS is about to arrive and therefore the deprecation of third-party cookies, it is advisable to check the code today in search of the point for which:

```
These cookies will not be restricted by cross-site cookie blocking since they cannot 
be used to track users' activity across different sites.
```
```
Change the setting to “Block Third-Party Cookies”.
...
Click on the request to “thirdparty.html” and click on the “Cookies” tab. 
You should see that the unpartitioned cookie was blocked in a third-party context,
but the partitioned cookie was allowed.
```
Context:
https://chromium.googlesource.com/website/+/718b6c16630e009bf7fd2629c933e9537aef06f9/site/updates/chips/index.md
https://bugs.chromium.org/p/chromium/issues/detail?id=1225444
https://bugs.chromium.org/p/chromium/issues/detail?id=1296161

it is advisable to check in priority where third-party cookies are blocked in the code, because it is possible that tomorrow it will be more complex with the tools available (read codesearch)

### Describe alternatives you have considered

I believe that knowing where the exact point in the code is important regardless of whether you want to activate CHIPS or not

**Comment by uazo** [comment]
maybe I found it:
"Differentiate partitioned and unpartitioned cross-site cookies in 3P"
https://source.chromium.org/chromium/chromium/src/+/86f47b025376bc76df06606b14cf39bd724d0571

**Comment by uazo** [comment]
the exact point seems to be

chromium side
[NetworkDelegateImpl::OnAnnotateAndMoveUserBlockedCookies](https://source.chromium.org/chromium/chromium/src/+/main:net/base/network_delegate_impl.cc;l=51;drc=3810f9d8c604ed06d6c14df979e940eabcc8c897;bpv=1;bpt=1) used by [URLRequestHttpJob::SetCookieHeaderAndStart](https://source.chromium.org/chromium/chromium/src/+/main:net/url_request/url_request_http_job.cc;l=727;drc=3810f9d8c604ed06d6c14df979e940eabcc8c897;bpv=1;bpt=1)

blink side (javascript)
[RestrictedCookieManager::GetAllForUrl](https://source.chromium.org/chromium/chromium/src/+/main:services/network/restricted_cookie_manager.cc;l=419;drc=3810f9d8c604ed06d6c14df979e940eabcc8c897;bpv=1;bpt=1?q=RestrictedCookieManager::GetAllForUrl&sq=&ss=chromium%2Fchromium%2Fsrc) used by [CookieStore::getAll](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/cookie_store/cookie_store.cc;l=282;drc=3810f9d8c604ed06d6c14df979e940eabcc8c897;bpv=1;bpt=1?q=CookieStore::getAll&sq=&ss=chromium%2Fchromium%2Fsrc) via mojom

from what I understand, the disabling flag of third-party cookies for now hardly will disappear, because it is used to disable local storage in third-party contexts, from since partitioning is not yet supported (as opposed to safari and firefox). I haven't found the corresponding bug yet.

There is a `kThirdPartyStoragePartitioning` flag (disabled) but I haven't seen how it works yet.
[here](https://source.chromium.org/chromium/chromium/src/+/main:components/browsing_data/core/features.cc) other things to understand. I think it is also necessary to understand if it is worth eliminating all cookie support `SameSite=None`

I keep checking ...

**Comment by csagan5** [comment]
Official blog post about CHIPS: https://developer.chrome.com/docs/privacy-sandbox/chips/

> it is advisable to check in priority where third-party cookies are blocked in the code, because it is possible that tomorrow it will be more complex with the tools available (read codesearch)

Bromite does not support any origin trial, so what do we exactly need to prepare for? At some point in time CHIPS will become enabled without origin trials; we can only decide to switch it on earlier or not.

What am I missing?

**Comment by uazo** [comment]
> what do we exactly need to prepare for? 
> At some point in time CHIPS will become enabled without origin trials
> What am I missing?

I am trying to evaluate if it is the case already now to activate it, with a patch that deactivates the function without modifying the trial.
I am not very convinced, but I am afraid that afterwards it will become more difficult to understand how to do it, and I wanted to mark the current points and review them at each release.
you know, without codesearch I would not be able to make any patch, and it seems to me that it only works with the master branch, but one day, I might need to see the code with the references of a version that is no longer available on cs . for this last point I should try https://sourcegraph.com sooner or later

**Comment by csagan5** [comment]
> if it is the case already now to activate it, with a patch that deactivates the function without modifying the trial.

Sorry, can't understand anything here. I said that CHIPS should at some point be active without origin trials. What is "activate it" and "deactivates the function" in your sentence? What is "without modifying the trial"? Trials don't work in Bromite.

> afterwards it will become more difficult to understand how to do it

How to do what? Please do not use "it", write down everything explicitly. The only thing to do is enable or disable CHIPS without origin trials (using a flag for example), can't think of anything else.

**Comment by uazo** [comment]
> What is "activate it" and "deactivates the function" in your sentence?

I don't remember exactly how to activate CHIPS in chromium, maybe simply activating the flag is enough or maybe we also need to activate the trial, I don't know.

> afterwards it will become more difficult to understand how - to do it - to disable CHIP

I meant that

> The only thing to do is enable or disable CHIPS (using a flag for example)

the flag today there is, but if the flag disappears from the code, I'm afraid that finding a way to disable it would become more complex

**Comment by csagan5** [comment]
> > The only thing to do is enable or disable CHIPS (using a flag for example)
> 
> the flag today there is, but if the flag disappears from the code, I'm afraid that finding a way to disable it would become more complex

I understand; if the origin trial ends and the flag is removed then more changes could be done without being gated by the flag. However, we never discussed if CHIPS should be disabled or not. Imagine a future where third-party cookies are gone and most sites use CHIPS: users might want to selectively have CHIPS for a site and not for another. In fact, for combination of sites and not for individual sites; this will pose a challenge for the current Chromium site settings interface.

I think eventually the upstream UI will evolve to support this; until that, if the flag is removed we can revert the commit and leave it disabled by default. And in worst case scenario we modify the cookie jar implementations to always return "not found" for any CHIPS site combination.

Makes sense?

**Comment by uazo** [comment]
> I understand; if the origin trial ends and the flag is removed then more changes could be done without being gated by the flag.

exact

> CHIPS should be disabled or not.

Chips are third party cookies isolated for firsty party.

>  users might want to selectively have CHIPS for a site and not for another.

which means to activate third-party cookies or not

> I think eventually the upstream UI will evolve to support this

why does chromium need a new UI? in my opinion the option to disable CHIPS will not even be introduced 

> ... return "not found" for any CHIPS site combination. Makes sense?

mah, who knows, we will have time to see it

**Comment by csagan5** [comment]
> > users might want to selectively have CHIPS for a site and not for another.
> 
> which means to activate third-party cookies or not, when google decides it's time to deprecate third-party cookies, as we know them now. the new third-party cookies are CHIPS, or any other name that the marketing will then decide CHIPS was wanted because it does not allow to track your browser between different top origin, but allows the tracking of the various sessions for the same site in third-party contexts, which is nowadays not allowed with the deactivation of third-party cookies.

Your point being? That it does not need any user option because with the introduction of CHIPS nobody should use this new type of third-party cookies anymore?

> > I think eventually the upstream UI will evolve to support this
> 
> why does chromium need a new UI? in my opinion the option to disable CHIPS will not even be introduced

To decide which third-party site groups are allowed or not, same thing I wrote earlier.

**Comment by uazo** [comment]
> That it does not need any user option because with the introduction of CHIPS nobody should use this new type of third-party cookies anymore

there is no need for any UI changes because there will be no option! Either active or inactive cookies, in my opinion, and that's it. but it is my opinion, not a fact.
how else could google deprecate the use of third-party cookies if they are actually so even though partitioned?

> To decide which third-party site groups are allowed

everyone will (1st + 3rd), or none, will choose the user, ok, but between those two options, I think

**Comment by csagan5** [comment]
See this section of the official blog post I mentioned: https://developer.chrome.com/docs/privacy-sandbox/chips/#why-its-important-to-opt-into-cookie-partitioning

Other browsers have already started trying things out. Even if there are no user options at the beginning there might be at some point after users request that; if that does not happen then Bromite has to get its own patch for it, or follow what Firefox or Safari do.

**Comment by csagan5** [comment]
@uazo what do we have to do to close this issue?

**Comment by uazo** [comment]
> what do we have to do to close this issue?

the idea is to double check it on every major release.
I saw that the introduction is scheduled for 2023, so there is no urgency it seems to me. if you want to close it go ahead, we will reopen it when necessary.

**Comment by uazo** [comment]
release announced: v109 (Jan 10, 2023)
https://groups.google.com/a/chromium.org/g/blink-dev/c/JNOQvsTxecI/m/gmIe7KCBAwAJ

**Comment by uazo** [comment]
time has come, in v109 google says it is [active by default](https://chromestatus.com/features#milestone%3D109).
fortunately reading the code doesn't seem to me, maybe they enable it with finch. the configuration (of finch) seems to be in the public domain, but I can't find it, do you know anything about it?

in any case, I think I am able to change the behaviour by enabling the partitioning of cookies by top site WITHOUT having to allow cross-site cookies. I'll let you know.
